Week3 : 전민주 스터디 필기, 과제 1 제출

mingmingmon opened the pull request:
Week3 : 전민주 스터디 필기, 과제 1 제출

mingmingmon committed:
Merge pull request #2 from KGU-C-Lab/master

10/10

mingmingmon committed:
Merge branch 'KGU-C-Lab:master' into master

mingmingmon committed:
Week3 : 전민주 스터디 필기, 과제 1 제출

Week3 : 전민주 스터디 필기, 과제 1 제출

mingmingmon commented:
Week3 : 전민주 스터디 필기, 과제 1 제출

mingmingmon committed:
Week3 : 전민주 과제2 제출

Week3 : 전민주 과제2 제출

mingmingmon commented:
Week3 : 전민주 과제2 제출

mingmingmon commented:
Week3 : 전민주 과제2 제출

chattymin commented on 전민주/Week 3/HW_1/app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,150 @@
+        android:text="SIGN UP"
+        android:textSize="40dp"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.5"
```
이 코드는 결과물에 영향을 주지 않기때문에 삭제하는게 좋을 것 같습니당

chattymin commented on 전민주/Week 3/HW_1/app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,150 @@
+        android:layout_marginTop="16dp"
+        android:text="About Me"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/editMBTI" />
```
@id와 @+id의 차이점은 조금 큽니다!
@+id는 새로 생성하는 것을 의미하기 때문에 @id로 코드 수정해주세용

chattymin commented on 전민주/Week 3/HW_2/app/src/main/java/com/mingmingmon/week3_hw_2/MainActivity.kt:
```diff
@@ -0,0 +1,110 @@
+            return num2
+        }
+
+        fun toastToPutNumbers(){ //값을 입력하지 않았을 때 toast로 값 입력해달라고 띄워주는 기능 만들고 싶어서 넣어봤습니다.
+            val text = "값을 입력해주세요"
+            Toast.makeText(
+                this,
+                text,
+                Toast.LENGTH_SHORT
+            ).show()
+        }
+
+        fun toastToWarnDivideZero(){
+            val text = "0으로 나눌 수 없습니다."
+            Toast.makeText(
+                this,
+                text,
+                Toast.LENGTH_SHORT
+            ).show()
+        }
```
토스트를 함수로 만든것 정말 좋은거 같습니다!!
근데 두 함수의 차이점이 내부 text라면, text를 변수로 받아오게 할 경우 하나로 줄일 수 있을 것 같네용

chattymin commented on 전민주/Week 3/HW_2/app/src/main/java/com/mingmingmon/week3_hw_2/MainActivity.kt:
```diff
@@ -0,0 +1,110 @@
+            val num1: Int? = getNum1()
+            val num2: Int? = getNum2()
+
+            if (num1 != null && num2 != null) {
+                val result: Int = num1 * num2
+                binding.resultText.text = "Result : $result"
+            }
+            else{
+                binding.resultText.text = "Result : -"
+                toastToPutNumbers()
+            }
```
이 부분도 중복이 많이 발생하는데 함수화 하면 더 좋을 것 같습니다!!
물론 지금도 정말 좋아요 :)

chattymin reviewed (changes requested):
이번 한주 고생하셨습니당 :)

mingmingmon committed:
week3 전민주 과제 refactor

refactor : 과제 1
- signup 페이지 구현
- constraintHorizontal_bias 코드 삭제
- 변수명 수정

refactor: 과제 2
-계산기 구현
-toast 기능 동작시키는 함수를 2개에서 1개로 수정. String으로 경고메세지 입력받고 출력할 수 있게 warningToast라는 함수 생성.
-+,-,*,/연산에서 중복되는 코드 when문으로 합침. calculate함수를 만들어서 when문 안에 배치
-변수명 수정

mingmingmon commented:
week3 전민주 과제 refactor

chattymin commented on 전민주/Week 3/HW_2/app/src/main/java/com/mingmingmon/week3_hw_2/MainActivity.kt:
```diff
@@ -0,0 +1,91 @@
+            return num2
+        }
+
+        fun warningToast(inputText: String) {
+            //String으로 상황에 맞는 경고 메세지를 받아서 띄우도록 수정했습니다.
```
재사용성 굿!!

chattymin commented on 전민주/Week 3/HW_2/app/src/main/java/com/mingmingmon/week3_hw_2/MainActivity.kt:
```diff
@@ -0,0 +1,91 @@
+        }
+
+        fun calculate(op: String): String {
+            val num1: Int? = getNum1()
+            val num2: Int? = getNum2()
+            val result: Int?
```
null 체크는 왜 한걸까요??

chattymin reviewed (approved):
완벽한 코드네요!! 너무 잘한다....
